natsstreaming/nats_streaming.go

```diff
@@ -61,6 +86,14 @@ func (p *asyncMessageSink) PublishMessages(ctx context.Context, acks chan<- subs
 
 	conn := p.sc
 
+	go func() {
```

Comment:
Why is this another goroutine, instead of reading from connectionLost in the select block at the bottom of the function.  Is it safe to write to rerr here without any synchronisation? What guarantees the write to rerr completes before the PublishMessages returns?

Reply:
fixed